packages/grant-explorer/src/context/BallotContext.tsx

```diff
@@ -85,7 +85,7 @@ export const useBallot = (): UseBallot => {
       (project) => project.projectRegistryId === projectToRemove.projectRegistryId
     );
 
-    const newShortlist = shortlist;
+    const newShortlist = [...shortlist];
```

Comment:
i think `const newShortlist = shortlist` doesn't copy the list contents -- so when `newShortlist.splice(...)` is called, it also updates the original `shortlist` list directly. Modifying state directly (outside of the `setState` function) is a big no-no in React, which might have been causing the bug :)

Fix is to make sure to create a copy, do any necessary changes/operations, then use `setState(updated copy)` to properly set the new state